contracts/mainnet/connectors/compound/v3/main.sol

```diff
@@ -0,0 +1,804 @@
+ */
+
+import { TokenInterface } from "../../../common/interfaces.sol";
+import { Helpers } from "./helpers.sol";
+import { Events } from "./events.sol";
+import { CometInterface } from "./interface.sol";
+
+abstract contract CompoundV3Resolver is Events, Helpers {
+	/**
+	 * @dev Deposit base asset or collateral asset supported by the market.
+	 * @notice Deposit a token to Compound for lending / collaterization.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be supplied. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param amt The amount of the token to deposit. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function deposit(
+		address market,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+
+		require(
+			market != address(0) && token != address(0),
+			"invalid market/token address"
+		);
+
+		bool isEth = token == ethAddr || token == wethAddr;
+		address token_ = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		if (token_ == getBaseToken(market)) {
+			require(
+				CometInterface(market).borrowBalanceOf(address(this)) == 0,
+				"debt-not-repaid"
+			);
+		}
+		if (isEth) {
+			amt_ = amt_ == uint256(-1) ? address(this).balance : amt_;
+			convertEthToWeth(isEth, tokenContract, amt_);
+		} else {
+			amt_ = amt_ == uint256(-1)
+				? tokenContract.balanceOf(address(this))
+				: amt_;
+		}
+		approve(tokenContract, market, amt_);
+
+		CometInterface(market).supply(token_, amt_);
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogDeposit(address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Deposit base asset or collateral asset supported by the market on behalf of 'to'.
+	 * @notice Deposit a token to Compound for lending / collaterization on behalf of 'to'.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be supplied. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE).
+	 * @param to The address on behalf of which the supply is made.
+	 * @param amt The amount of the token to deposit. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function depositOnBehalf(
+		address market,
+		address token,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+
+		require(
+			market != address(0) && token != address(0),
+			"invalid market/token address"
+		);
+		require(to != address(0), "invalid to address");
+
+		bool isEth = token == ethAddr || token == wethAddr;
```

Comment:
```suggestion
		bool isEth = token == ethAddr;
```

Reply:
@Richa-iitr Why this? ` || token == wethAddr`

Reply:
done